examples/twittermap/web/public/javascripts/sidebar/controllers.js

```diff
@@ -11,6 +11,58 @@ angular.module("cloudberry.sidebar", ["cloudberry.common"])
 
     $scope.currentTab = "aboutTab";
 
+    // Timer for sending query to check whether it can be solved by view
+    $scope.timerCheckQuerySolvableByView = null;
+
+    // queryID used to identify a query, which is sent by timer
+    $scope.nowQueryID = null;
+
+    // A WebSocket that send query to Cloudberry, to check whether it is solvable by view
+    var wsCheckQuerySolvableByView = new WebSocket(cloudberryConfig.checkQuerySolvableByView);
+
+    function closeRightMenu() {
```

Comment:
There's a small bug here: this `closeRightMenu` function needs to include the operations of setting `isHashTagOpen` and `isSampleTweetsOpen` flags to be `false` as well. Otherwise, although you close the sidebar, the `hashtag` or `sample tweets` query will still be sent to Cloudberry.

Reply:
For open/ close sidebar, I find a bug in current code.
![image](https://user-images.githubusercontent.com/26839941/45729644-097cdd00-bc00-11e8-92b7-0178a25d4146.png)

In the code, we use different button for open and close sidebar separately. 
- When click `hambugerButton`, it opens sidebar, and changes the flags by `showOrHideSidebar` function.
- When click `closebtn`, it only closes sidebar, but **not** change the flags.

I think it is because we use only one button for both open and close in previous version. 
Now we use two button, we need to change the code.

Reply:
Should I fix the bug in this PR?

Reply:
Oh, I see. It was introduced by the `hamburger` button. Let's fix it since it's just one line fix. I think just adding the `ng-click="showOrHideSidebar(-1)"` will solve the problem. Please correct me if I'm wrong.

Reply:
And we don't need `click = click * -1` anymore. Since open and close buttons are separated.

Reply:
I don't think we need variable `click` anymore. So I remove the code about click. 
Also, I move the js code from `index.scala.html` to `sidebar/controller.js` to avoid duplicate function.